src/relayclient/AccountManager.ts

```diff
@@ -43,34 +54,52 @@ export default class AccountManager {
   // TODO: make forwarder part of RelayRequest, why is it dangling??
   async sign (relayRequest: RelayRequest, forwarderAddress: Address): Promise<PrefixedHexString> {
     let signature
-    const signedData = await getDataToSign({
+    const signedData = getDataToSign({
       chainId: this.chainId,
       verifier: forwarderAddress,
       relayRequest
     })
     const keypair = this.accounts.find(account => isSameAddress(account.address, relayRequest.relayData.senderAddress))
     if (keypair != null) {
-      // @ts-ignore
-      signature = sigUtil.signTypedData_v4(this.ephemeralKeypair.privateKey, { data: signedData })
+      signature = this._signWithControlledKey(keypair, signedData)
     } else {
-      signature = await getEip712Signature(
-        {
-          web3: this.web3,
-          methodSuffix: this.config.methodSuffix ?? '',
-          jsonStringifyRequest: this.config.jsonStringifyRequest ?? false,
-          dataToSign: signedData
-        })
+      signature = await this._signWithExternallyControlledKey(signedData)
+    }
+
+    // Sanity check
+    let rec: Address
+    try {
+      // @ts-ignore
+      rec = sigUtil.recoverTypedSignature_v4({
+        data: signedData,
+        sig: signature
+      })
+    } catch (error) {
+      // eslint-disable-next-line @typescript-eslint/restrict-template-expressions
+      throw new Error(`Failed to sign relayed transaction for ${relayRequest.relayData.senderAddress}`)
     }
-    // Sanity check only
-    // @ts-ignore
-    const rec = sigUtil.recoverTypedSignature_v4({
-      data: signedData,
-      sig: signature
-    })
     if (!isSameAddress(relayRequest.relayData.senderAddress.toLowerCase(), rec)) {
       // eslint-disable-next-line @typescript-eslint/restrict-template-expressions
```

Comment:
better disable restrict-template-expressions globally. we never have templates, and have to add this each time.

Reply:
You only need to add it if TypeScript does not know what type your expression has. This is a symptom, I do not want to globally silence the symptom.
Some libraries won't have any type annotations, and this is ok, as long as we mark those explicitly.

Reply:
we use backtick notation generously, not only for template values.
(e.g. if you put an expression, it will complain, because it expects a single-var name for template values)